src/it/java/io/github/jpmorganchase/fusion/packaging/DatasetOperationsIT.java

```diff
@@ -162,6 +163,108 @@ public void testCreateDatasetOfTypeReport() {
         //TODO :: Contract for response of dataset.create() needs to be decided
     }
 
+    @Test
+    public void testCreateDatasetOfTypeFlowInput() {
+        // Given
+        wireMockRule.stubFor(WireMock.post(WireMock.urlEqualTo("/catalogs/common/datasets/SIF0001"))
+                .withRequestBody(equalToJson(TestUtils.loadJsonForIt("dataset/dataset-flow-SIF0001-create-request.json")))
+                .willReturn(WireMock.aResponse()
+                        .withHeader("Content-Type", "application/json")
+                        .withStatus(200)
+                        .withBodyFile("dataset/dataset-create-response.json")));
+
+        Dataset dataset = getSdk().builders().dataset()
+                .identifier("SIF0001")
+                .description("Sample input flow description 1")
+                .linkedEntity("SIF0001/")
+                .title("Sample Input Flow 1 | North America")
+                .frequency("Daily")
+                .varArg("category", listOf("Category 1"))
+                .varArg("createdDate", "2022-02-06")
+                .varArg("coverageStartDate", "2022-02-06")
+                .varArg("coverageEndDate", "2023-03-09")
+                .varArg("isThirdPartyData", Boolean.FALSE)
+                .varArg("isInternalOnlyDataset", Boolean.FALSE)
+                .varArg("language", "English")
+                .varArg("maintainer", "Maintainer 1")
+                .varArg("modifiedDate", "2023-03-09")
+                .varArg("publisher", "Publisher 1")
+                .varArg("region", listOf("North America"))
+                .varArg("source", listOf("Source System 1"))
+                .varArg("subCategory", listOf("Subcategory 1"))
+                .varArg("tag", listOf("Tag1"))
+                .varArg("isRestricted", Boolean.FALSE)
+                .varArg("isRawData", Boolean.FALSE)
+                .varArg("hasSample", Boolean.FALSE)
+                .applicationId(Application.builder().sealId("12345").build())
+                .flow(
+                        Flow.builder()
+                                .flowDirection("Input")
+                                .producerApplicationId(Application.builder().sealId("123456").build())
+                                .consumerApplicationId(new Application[]{Application.builder().sealId("456789").build()})
```

Comment:
Another tweak if we may ... To avoid the client having to manually create an Application[], use the builder on the Flow to make this easier. I.e

```
Flow.builder()
.producerApplicationId(Application.builder().sealId("12345").build()
.consumerApplicationId(Application.builder().sealId("11111").build())
.consumerApplicationId(Application.builder().sealId("22222").build())
.build()
```

Reply:
change is incorporated now